dataclients/kubernetes/kube.go

```diff
@@ -310,12 +312,13 @@ func (c *Client) getServiceURL(namespace, name string, port backendPort) (string
 }
 
 // TODO: find a nicer way to autogenerate route IDs
-func routeID(namespace, name, host, path string) string {
+func routeID(namespace, name, host, path, backend string) string {
 	namespace = nonWord.ReplaceAllString(namespace, "_")
 	name = nonWord.ReplaceAllString(name, "_")
 	host = nonWord.ReplaceAllString(host, "_")
 	path = nonWord.ReplaceAllString(path, "_")
-	return fmt.Sprintf("kube_%s__%s__%s__%s", namespace, name, host, path)
+	backend = nonWord.ReplaceAllString(backend, "_")
+	return fmt.Sprintf("kube_%s__%s__%s__%s__%s", namespace, name, host, path, backend)
```

Comment:
the route based ones, yes. However, it should be possible to collect them the old way, because the metrics endpoint does a prefix match on them. Is that an option?